### PR TITLE
[wip] Feature: projects, adds a 'bioprotocol' project

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

File: projects/elife.yaml [hunk 1]
```diff
@@ -2169,3 +2169,27 @@ microsimulation-demo:
...
+            - 22
+            - 80
+            - 443
+        sns:
+            # endpoint for BioProtocol to publish updates to article protocol data
+            - bioprotocol-updates--{instance}
+        sqs:
+            # single queue that receives messages about:
+            # - updates to articles (internally)
+            # - updates to article protocol data (from BioProtocol)
```
Comment (giorgiosironi) [inline projects/elife.yaml]: these will be wildly different messages, but if the application is smart enough

File: projects/elife.yaml [hunk 2]
```diff
@@ -2169,3 +2169,27 @@ microsimulation-demo:
...
+    repo: https://github.com/elifesciences/bioprotocol-service
+    aws:
+        type: t2.micro
+        ports:
+            - 22
+            - 80
+            - 443
+        sns:
+            # endpoint for BioProtocol to publish updates to article protocol data
+            - bioprotocol-updates--{instance}
```
Comment (giorgiosironi) [inline projects/elife.yaml]: https://github.com/elifesciences/issues/issues/4930#issuecomment-498144947 states they would be writing to SQS; it doesn't make much of a difference but has this changed?

File: projects/elife.yaml [hunk 3]
```diff
@@ -2169,3 +2169,27 @@ microsimulation-demo:
         ports:
             1280: 80
             1281: 8081
+
+bioprotocol:
+    repo: https://github.com/elifesciences/bioprotocol-service
```
Comment (giorgiosironi) [inline projects/elife.yaml]: no formula mentioned? This influences some `builder` behavior, I think

File: projects/elife.yaml (hunk 2 above)
Comment (lsh-0) [reply projects/elife.yaml]: I thought that was a mistake. Writing to a queue doesn't seem correct to me. Writing to a topic and having queues subscribed to those topics seems more-correct. Or I could just write an API endpoint they can PUT/POST to directly. I'll follow this point up with them

File: projects/elife.yaml (hunk 1 above)
Comment (lsh-0) [reply projects/elife.yaml]: dispatching a message based on the shape of the data seems like a pretty basic thing to do.

File: projects/elife.yaml (hunk 3 above)
Comment (lsh-0) [reply projects/elife.yaml]: I hadn't gotten to that, I'll do it today.

Comment (lsh-0) [reply projects/elife.yaml]: done

File: projects/elife.yaml (hunk 2 above)
Comment (lsh-0) [reply projects/elife.yaml]: I've contacted Dong. He seems like a cheerful guy.

Comment (giorgiosironi) [reply projects/elife.yaml]: Queues (SQS) are meant for 1-to-1 communication, whereas topics/fanouts (SNS) are meant for 1-to-N communication.

Comment (lsh-0) [reply projects/elife.yaml]: I know, but that seems so very limited to almost be a side effect of the larger design AWS has with SNS+SQS. There is an awful lot I can do while I wait for Dong to confirm so no urgency.

Comment (giorgiosironi) [reply projects/elife.yaml]: With different names, it is a general pattern of messaging systems:
https://www.enterpriseintegrationpatterns.com/patterns/messaging/PointToPointChannel.html
https://www.enterpriseintegrationpatterns.com/patterns/messaging/PublishSubscribeChannel.html

Comment (lsh-0) [reply projects/elife.yaml]: yes, thank you Giorgio. (this is getting a little condescending, please stop.)

Comment (lsh-0) [reply projects/elife.yaml]: they're happy to use POST. I've removed the SNS topic and SQS subscription

**Pull request merged**